### PR TITLE
Ux enhancements - explanatory info on lock and language changes

### LINKED ISSUE #915: Add toast to notify user that input is locked
- [x] Toast when trait layout area is touched when locked
- [x] Toast describing each mode as the locked icon is toggled
- [x] Add bonk sound when interacting with UI when locked

### PR DESCRIPTION
**Pull request by bellerbrock**
## Description

_Provide a summary of your changes including motivation and context.
If these changes fix a bug or resolves a feature request, be sure to link to that issue._

Adds toasts when locked status is changed and toast plus error sound when locked interface is clicked.
Adds dialog when language is switched to warn that translations are crowd-sourced and may be incomplete  
Closes #915 and #890

## Type of change

_What type of changes does your code introduce? Put an `x` in boxes that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated relevant documentation

## Changelog entry

_Please add a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note
Explanatory info added to lock and languages changes
```

### CODE REVIEW

**Comment by trife** [comment]
- [ ] Text traits are still editable
- [ ] Dismiss toasts before displaying new ones

**Comment by chaneylc** [comment]
I fixed the text trait locking issue, tested with scanners too.

File: app/src/main/res/layout/activity_collect.xml [hunk 1]
```diff
@@ -143,4 +143,16 @@
...
+        android:id="@+id/lockOverlay"
+        android:layout_width="0dp"
+        android:layout_height="0dp"
+        android:background="@android:color/transparent"
+        android:visibility="gone"
+        android:clickable="true"
+        app:layout_constraintTop_toTopOf="@id/act_collect_input_view"
+        app:layout_constraintBottom_toBottomOf="@id/toolbarBottom"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintEnd_toEndOf="parent" />
```
Comment (chaneylc) [inline app/src/main/res/layout/activity_collect.xml:156]: I originally went with implementing locking like this. I discussed this with @trife when I realized it disables scrolling and viewing captured images. Now it also disables repeated measures, and the metadata dialog. @trife let me know if this is what you wanted but originally we decided against this approach for these reasons. This approach really does not pair well with the existing repeated measures, unless we maybe constrain it below the collect input view when repeated measures is enabled.

Comment (trife) [reply app/src/main/res/layout/activity_collect.xml:156]: Let's try this approach and see if we get push back from users. The other approach seems more prone to error.

**Pull request merged**